### PR TITLE
Adopt OMR's Recognized Call Transformer

### PR DESCRIPTION
**Pull request by 0dvictor**
Initial adoption of OMR's Recognized Call Transformer;
converting certain recognized calls to IL OpCode on X86.

Signed-off-by: Victor Ding <dvictor@ca.ibm.com>

### CODE REVIEW

File: runtime/tr.source/trj9/optimizer/RecognizedCallTransformer.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+* Copyright (c) 2000, 2017 IBM Corp. and others
```
Comment (andrewcraik) [inline runtime/tr.source/trj9/optimizer/RecognizedCallTransformer.cpp]: dates need fixing

**Comment by andrewcraik** [comment]
The idea is good, but we need a better design in OMR of the opt pass and the extension points so that OpenJ9 doesn't implement the whole walk - OMR should do the walk and delegate checking to OpenJ9.

**Comment by 0dvictor** [comment]
Good suggestion, let me move the walk to OMR.

**Comment by andrewcraik** [comment]
Jenkins test sanity depends eclipse/omr#1836

**Comment by andrewcraik** [comment]
Jenkins test sanity depends eclipse/openj9-omr#7

**Comment by andrewcraik** [comment]
Jenkins test sanity depends eclipse/openj9-omr#7

**Comment by andrewcraik** [comment]
Jenkins test sanity depends eclipse/openj9-omr#7

**Comment by fjeremic** [comment]
FYI this is currently blocking eclipse/openj9-omr promotion:
https://ci.eclipse.org/openj9/job/Pipeline-OMR-Acceptance/

**Comment by 0dvictor** [comment]
Both sanity tests passed. CI test failed because the required OMR changes has not been promoted into OpenJ9.

**Comment by 0dvictor** [comment]
Once this PR merged, OpenJ9 should be able to upgrade to latest OMR.

**Pull request merged**